core/src/main/java/org/wso2/testgrid/core/command/GenerateReportCommand.java

```diff
@@ -75,12 +78,40 @@ public void execute() throws CommandExecutionException {
                     "\tProduct version: " + productVersion + "\n" +
                     "\tChannel" + channel);
 
+            Product product = getProduct(productName, productVersion, channel);
             TestReportEngine testReportEngine = new TestReportEngine();
-            testReportEngine.generateReport(productName, productVersion, channel, showSuccess, groupBy);
+            testReportEngine.generateReport(product, showSuccess, groupBy);
         } catch (ReportingException e) {
             throw new CommandExecutionException(StringUtil
                     .concatStrings("Error occurred when generating test report for { product: ", productName,
                             ", version: ", productVersion, ", channel: ", channel, " }"), e);
         }
     }
+
+    /**
+     * Returns an instance of {@link Product} for the given product name and product version.
+     *
+     * @param productName    product name
+     * @param productVersion product version
+     * @param channel        product test plan channel
+     * @return an instance of {@link Product} for the given product name and product version
+     * @throws CommandExecutionException throw when error on obtaining product for the given product name and product
+     *                                   version
+     */
+    private Product getProduct(String productName, String productVersion, String channel)
+            throws CommandExecutionException {
+        try {
+            ProductUOW productUOW = new ProductUOW();
+            Product.Channel productChannel = Product.Channel.valueOf(channel);
+            return productUOW.getProduct(productName, productVersion, productChannel)
+                    .orElseThrow(() -> new CommandExecutionException(StringUtil
+                            .concatStrings("No product test plan found for product {product name: ", productName,
+                                    ", product version: ", productVersion, ", channel: ", channel, "}")));
```

Comment:
This will be logged in [1]. So when the exception is thrown we know what it caused and the action performed.

[1] https://github.com/vidurananayakkara/wso2-test-grid/blob/a508c35a2f3d928681b8b239c4ea39946cb2a592/core/src/main/java/org/wso2/testgrid/core/command/GenerateReportCommand.java#L85

Reply:
Ah.. I do not see remedial information in there though. May be I'm missing something.